rpc.go

```diff
@@ -309,7 +309,10 @@ func (t *Topic) listen() {
 
 func (t *Topic) publishResponse(from peer.ID, id cid.Cid, data []byte, e error) {
 	topic, err := newTopic(t.ctx, t.ps, t.host, responseTopic(t.t.String(), from), false)
-	if err != nil {
+	if err != nil &&
+		// this can happen when more than one messages are received
+		// from the same peer.
+		!strings.Contains(err.Error(), "topic already exists") {
```

Comment:
A decision also needs to be made here: Should we continue or simply return if topic exists? For our usage, an existing topic means duplicated message, which should simply be ignored (the only topic with more than one messages is the AuctionTopic, which doesn't watch for response), but for general user case, it should be totally fine to send many messages on a single topic and expect many responses.

I vote for correctness of this package and let the caller handle duplications

Reply:
Oop continuing won't work... as `top` is `nil` for whatever `err` returned.

Reply:
this must happen when the first response topic hasn't been closed yet. I think it's reasonable to assume only one response can be sent at a time... but many can be sent over the lifetime of the topic.

conclusion: just ignore the error and return from `publishResponse`.

Reply:
Again, this relates to race condition - because `publishResponse` runs in a separate goroutine, what if there are actually more than one messages arrive, hence start several goroutines? All but the first one would be responded.

Even if we respond all messages, they can be out of order, hence a bigger problem: the other end gets response for a wrong message. The simplest would be to stop spawning new goroutine for response.